components/menu/menu-item-styles.js

```diff
@@ -9,13 +9,17 @@ export const menuItemStyles = css`
 		box-sizing: border-box;
 		color: var(--d2l-color-ferrite);
 		cursor: pointer;
-		display: block;
+		display: none;
 		font-size: 0.8rem;
 		margin-top: -1px;
 		outline: none;
 		width: 100%;
 	}
 
+	:host([visible]) {
```

Comment:
Hmmmm, this seems a bit odd.  🤔 We set `display: none` by default, but then we have a `visible` property that applies `flex` (is that valid for all menu items?), but we also have styles for `hidden` property that sets `none`.

Reply:
So I added this because if a menu-item (e.g., custom-menu-item) is imported before the menu in firefox, then it will render early and there's a flash of just that menu item with not-so-great style (e.g., lone menu item taking up max width). I think this should be `block` though. You're right `visible` is a bit weird. Maybe something like `ready` would be better if this case is necessary to deal with.

Reply:
I _think_ there's just this left and then it looks good to merge.  😄

Reply:
So just to clarify this - this is only applicable on edge when a custom menu item is used and what will happen (if this is not used) is that we will see the custom menu item only for a second and then the other menu items will load. With this added then all the menu items are shown when ready, at the same time.
Options:
- Rename this to `ready` (or something other than `visible`) and fix up so `block` instead of `flex`, remove once edge is chredge
- Remove this and be okay with the flash of lone menu item (which in my opinion isn't terrible and probably fairly rare)

Reply:
I don't see a flash in the demo page when using Edge.  Is it noticeable for you with the demo or is it something you noticed elsewhere? My inclination (without seeing the flash) is to not do this, but if we do, I would also make the property private so it's extra clear that it's not part of the API that consumers would try to set, and then also call it out as Edge only so we can easily spot & remove it later.

It is interesting that originally the default was `block`, but `d2l-menu-item` overrode it to be `flex`. We'd probably want that restored if we remove this.